test/Altinn.Notifications.Tests/Notifications.Integrations/ShortMessageService/ShortMessageServiceClientTests.cs

```diff
@@ -0,0 +1,243 @@
+            PropertyNameCaseInsensitive = true,
+            DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
+        };
+
+        var delegatingHandler = new DelegatingHandlerStub((request, token) =>
+        {
+            if (!request!.RequestUri!.AbsolutePath.EndsWith("instantmessage/send"))
+            {
+                return Task.FromResult(new HttpResponseMessage(HttpStatusCode.NotFound));
+            }
+
+            ShortMessage? message = null;
+            try
+            {
+                string content = request.Content!.ReadAsStringAsync(token).GetAwaiter().GetResult();
+                message = JsonSerializer.Deserialize<ShortMessage>(content, serializerOptions);
+            }
+            catch
+            {
+                /* Ignore deserialization errors for this test */
+            }
+
+            if (message == null)
+            {
+                return Task.FromResult(new HttpResponseMessage(HttpStatusCode.BadRequest)
+                {
+                    Content = new StringContent("Invalid JSON format", Encoding.UTF8, "application/json")
+                });
+            }
+
+            HttpStatusCode statusCode;
+            string? errorContent = null;
+            switch (message.Recipient)
+            {
+                case "client-closed":
+                    statusCode = (HttpStatusCode)499;
+                    errorContent = "{\"title\":\"Request terminated\",\"status\":499,\"detail\":\"Request was canceled before processing could complete\"}";
+                    break;
+
+                case "network-error":
+                    throw new HttpRequestException("Service unavailable", null, HttpStatusCode.ServiceUnavailable);
+
+                case "generic-error":
+                    throw new InvalidOperationException("Some unexpected error occurred");
+
+                default:
+                    if (MobileNumberHelper.IsValidMobileNumber(message.Recipient))
+                    {
+                        statusCode = HttpStatusCode.OK;
+                    }
+                    else
+                    {
+                        statusCode = HttpStatusCode.BadRequest;
+                        errorContent = "{\"title\":\"Invalid request format\",\"status\":400,\"detail\":\"The phone number format is invalid\"}";
+                    }
+
+                    break;
+            }
+
+            var response = new HttpResponseMessage(statusCode);
+            if (errorContent != null)
+            {
+                response.Content = new StringContent(errorContent, Encoding.UTF8, "application/json");
+            }
+
+            return Task.FromResult(response);
+        });
+
+        PlatformSettings settings = new()
+        {
+            ApiShortMessageServiceEndpoint = "http://localhost:5092/notifications/sms/api/v1/"
+        };
+
+        return new ShortMessageServiceClient(new HttpClient(delegatingHandler), _loggerMock.Object, Options.Create(settings));
```

Comment:
For optimization, consider instantiating the platformsettings options statically in the test class rather than new'ing it up and doing `Options.Create` on every run of CreateShortMessageServiceTestClient, 
```suggestion
        return new ShortMessageServiceClient(new HttpClient(delegatingHandler), _loggerMock.Object, _platformSettings);
```